File: src/main/java/de/pflugradts/passbird/domain/service/password/storage/PasswordStoreAdapterPort.java
package de.pflugradts.passbird.domain.service.password.storage;

import de.pflugradts.passbird.domain.model.password.PasswordEntry;

import java.util.function.Supplier;
import java.util.stream.Stream;

/**
 * AdapterPort for syncing the Password Repository with and restoring it from a physical file.
 */
public interface PasswordStoreAdapterPort {
    Supplier<Stream<PasswordEntry>> restore();
    void sync(Supplier<Stream<PasswordEntry>> passwordEntries);
}

File: src/main/kotlin/de/pflugradts/passbird/domain/service/password/storage/PasswordStoreAdapterPort.kt
package de.pflugradts.passbird.domain.service.password.storage

/**
 * AdapterPort for syncing the Password Repository with and restoring it from a physical file.
 */
interface PasswordStoreAdapterPort {
    fun restore(): EggStreamSupplier
    fun sync(supplier: EggStreamSupplier)
}
